README.md

```diff
@@ -1,22 +1,122 @@
 # Ethereum Alarm Clock
 
 [![Join the chat at https://gitter.im/pipermerriam/ethereum-alarm-clock](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/pipermerriam/ethereum-alarm-clock?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+[![Build Status](https://travis-ci.org/chronologic/ethereum-alarm-clock.svg?branch=master)](https://travis-ci.org/chronologic/ethereum-alarm-clock)
 
 Source code for the [Ethereum Alarm Clock service](http://www.ethereum-alarm-clock.com/)
 
+## What is the EAC (Ethereum Alarm Clock)
+
+The Ethereum Alarm Clock is a smart contract protocol for scheduling Ethereum transactions to be executed in the future. It allows any address to set the parameters of a transaction and allow clients to call these transactions during the desired window. The EAC is agnostic to callers so can be used by both human users and other smart contracts. Since all of the scheduling logic is contained in smart contracts, transactions can be scheduled from solidity.
+
+Additionally the EAC faciliates the execution of this pool of scheduled transactions through a client. The EAC client continuously runs and searches for transactions which are scheduled to be executed soon then claims and executes them. For the EAC to be successful it depends on users who run execution clients. There are a few ways incentives for running these execution clients are baked in to the protocol itself, notably the claiming mechanism and the reward payment. 
 
 ## Running the tests
 
-Testing for Alarm is done with [populus](https://github.com/pipermerriam/populus) which can be installed with `pip`.
+_Tests have been ported to Javascript and can now be run using the Truffle Suite_
+
+Originally the test suite was written in Python using the Populus framework, these still exist for reference under the populus-tests/ directory. However, we have ported over the suite to use the Truffle framework since this may be more familiar to developers who know the Ethereum tooling in Javascript. These tests can be found in the test/ directory.
+
+If you would like to run the test please set up your environment to use node v8.0.0, truffle v4.0.1 and the latest ganache-cli.
 
-```shell
-$ pip install populus
 ```
+nvm use 8.0.0
+npm i
+npm i -g truffle@4.0.1 
+npm i -g ganache-cli
+```
+
+Start ganache-cli in a terminal screen by running `ganache-cli`.
+
+In another terminal screen run `npm test` at the root of the directory. This will run the npm test script that splits up the tests into different runtimes. The tests are split because the EAC is a moderately sized project and running all the tests with one command has a tendency to break down the ganache tester chain.
+
+Each time you run the tests it is advised to rebuild your build/ folder, as this may lead to bugs if not done. You can do this by running the command `rm -rf build/`.
+
+## Documentation
+
+Documentation can be found [here](https://eac-docs.surge.sh/).
```

Comment:
Any reason you're using `surge.sh` over readthedocs?

Reply:
I wasn't familiar with using readthedocs, but it's super easy. I migrated the docs to build there instead.